src/common/buffer.cc

```diff
@@ -777,6 +781,38 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
   unsigned buffer::ptr::raw_length() const { assert(_raw); return _raw->len; }
   int buffer::ptr::raw_nref() const { assert(_raw); return _raw->nref.read(); }
 
+  void buffer::ptr::copy_out(unsigned o, unsigned l, char *dest) const {
+    assert(_raw);
+    if (o+l > _len)
+        throw end_of_buffer();
+    char* src =  _raw->data + _off + o;
+    if (l > 8) {
+        memcpy(dest, src, l);
+        return;
+    }
+    switch (l) {
+        case 8:
+            *((uint64_t*)(dest)) = *((uint64_t*)(src));
```

Comment:
What happens if dest is not word aligned?  Are there some architectures hwere it'll fail outright?

If it's slower, perhaps the if (l > 8) should be if (l > 8 || (dest & 0x7)) ?